File: 1071.js
const fs = require('fs');

function solve() {
	const input = fs.readFileSync('/dev/stdin').toString().split('\n');
	let inputIdx = 0;

	const N = +input[inputIdx++];
	const arr = input[inputIdx++]
		.split(' ')
		.map((v) => +v)
		.sort((a, b) => a - b);

	let idx = 0;

	while (idx < N) {
		if (arr[idx] === arr[idx - 1] + 1) {
			let start = idx - 1,
				end = idx + 1;
			for (; start >= 1 && arr[start - 1] === arr[idx - 1]; start--);
			for (; end < N && arr[end] === arr[idx]; end++);
			if (end === N) {
				const before = arr[idx - 1],
					after = arr[idx];
				const midIdx = start + (end - idx);
				for (let i = start; i < midIdx; arr[i++] = after);
				for (let i = midIdx; i < end; arr[i++] = before);
			} else {
				[arr[idx], arr[end]] = [arr[end], arr[idx]];
			}
			idx = end + 1;
		} else {
			idx++;
		}
	}
	console.log(...arr);
}

solve();

File: 1071.py
from sys import stdin

input = stdin.readline


def solve():
    N = int(input())
    arr = sorted([*map(int, input().split())])
    i = 1

    while i < N:
        if arr[i] == arr[i-1] + 1:
            start, end = i-1, i+1
            while start >= 1 and arr[start-1] == arr[i-1]:
                start -= 1
            while end < N and arr[end] == arr[i]:
                end += 1
            if end == N:
                arr[start:end] = [arr[i]] * (end - i) + [arr[i-1]] * (i - start)
                i = end
            else:
                arr[i], arr[end] = arr[end], arr[i]
                i = end + 1
        else:
            i += 1

    print(*arr)


if __name__ == '__main__':
    solve()
